### ISSUE TITLE
The laser head appears to be randomly moving to other parts of the bed between cuts

### ISSUE THREAD

**Issue by JackiePease** [DoESLiverpool/somebody-should#1503]
Kat was unable to do any work this morning due to the he laser head jumping between cuts. Fatih was having a similar problem on Monday - @MatthewCroughan believed this was due to saving with a different version of AutoCAD - I think they were able to produce a successful result using v14?

Kat was using Inkscape and Corel Draw though. Rob White tried several different ways to get round the problem but none worked. She says she'll email in her file so we can investigate later today.

**Comment by MatthewCroughan** [comment]
A lot of this will surely be solved when we move all the software over to Lightburn, Laserweb or anything else that can interface with GRBL. This is a possibility once we have completed #955 as we can upgrade the other cutters.

I do not believe this is an issue worth spending much time on. We really should just throw money at this problem and afford the new hardware and its installation so we can use better software with the cutters.

**Comment by amcewen** [comment]
Except it isn't just a case of "throw[ing] money at this problem".  There's also the downtime while the upgrade happens (and the @DoESLiverpool/laser-maintenance team have a list of other things they'd like to do at the same time) to balance against having no working cutter.  

Plus there's also the task of retraining everyone who uses the cutter on the new software.  The current plan is to run Tony alongside Gerald for a while to allow a transition period rather than a hard bottleneck of "everyone needs a new induction" at once.

Do you have any info to add about how you and Faith got on on Monday, @MatthewCroughan, given that's what @JackiePease actually asked about?

**Comment by MatthewCroughan** [comment]
@amcewen I commented on what I think the quicker, more pragmatic solution is. Thank you for disagreeing :D

Obviously this upgrade would have to happen overnight and be usable.

I think the new software (lightburn) will be instantly accessible to people that are familiar with Laser Cut 5.3. It'll be much preferable to the job ending bugs that cost people time and money when using the space. Aka a better situation, don't you think they'll be relieved?

I watched Rob White battle with these bugs for 6 months last year. It cost him money and time.

If now is not a good time to do it, then I don't know when is. If retraining is as big an issue as you state, then we will remain this way for quite a long time.

**Comment by JackiePease** [comment]
@Sean-anotherone believes this is a hardware issue - will know more after this evening

**Comment by MatthewCroughan** [comment]
@JackiePease it will be good to make sure. My comments are more broad and reference not only this bug, but other bugs and usability problems that are outside the scope of this issue, such as being unable to unite the lines of some complex geometry, import certain polylines, work with svgs, layers going missing, etc.

Worst of all, not crashing or reporting an error when unsupported dxf features are present, or failing to report a missing font when fonts are present in the file, insidiously corrupting your cut job and sending the laser flying and doing strange things.

**Comment by FatihDeveci** [comment]
@amcewen I have been facing different problems for several months. I will try to explain them in detail below. The ranking goes from small problems to big ones.

1- Sometimes some geometries in the drawing file are shifted about 5cm from where they should be. Sometimes they are shifted about 0.5cm. (In the drawing file, all the geometries are exactly in the right place). Breaking down some polylines can be a temporary solution.  However, errors still persist in lots of files.

2- In some files, some layers are completely shifted. For example, the "Draw" layer on the material is in the right place, but the "Cut" layers can sometimes shift about 1cm, sometimes 5cm. However, the problem is not related to the layers, as there are the same errors when the whole drawing is on the same layer.


3- In some trials, the laser head starts to go and come and go on a line of a few mm at the same place. It is necessary to stop the laser cutter and cancel the cutting so that the material does not burn.  I had this error at the very beginning of a file on Monday. Then I took the whole drawing on a single layer and randomly changed the color of that layer. This time it started cutting, but after 1 minute the drawings started to shifting again.

4- The most important part is this: When the drawings are shifted, the starting point of the laser machine is also shifted. When the cutting is completed, instead of returning to the starting point, the laser head sometimes ends 5cm away, sometimes 20cm away, sometimes at a completely different point. This shifting occurs while the machine is cutting. At that shifting moment, there is a big noise as if the laser head is stuck somewhere. After that noise, all the drawings start to be shifted.

5- On Monday, when the cutting of a vertical A3 size drawing file was completed, noise came from the laser again. Since I cut the same file 5 times, I carefully observed the same event 5 times. I focused the laser head on the upper end of the paper material. However, when the cut was finished, first the big noise and then the finishing beep sound came. When the cut was finished, the laser head was 40cm lower on y-axis than it should have been.Then I moved the laser head up again on the y-axis to cut the same file again. When I pressed the "Test" button, the laser head defined a horizontal A3 or other longer area instead of vertical A3. I tried 5 times with the same file and the same thing happened each time. So when the cut was finished, the file on the machine was changed, I downloaded the same file from the computer again to cut the true file again.

6- While cutting large-sized materials in the past months, the laser head started to get stuck somewhere in the middle of the bed, trying to cut the same small line forever. After trying a few times, I started placing my material away from that area. It's kind of a Bermuda Triangle for me :) However, replacing the rail and faulty bearing a few months ago might have solved this, I'm not sure.

These problems make me think the problem is hardware related. Unfortunately I cannot use the laser cutter for these reasons. These problems have been recurring for a long time (months), but I've done a lot of trials for 4 hours on Monday and haven't been able to finish my cuts.

Unlike these, a non-critical problem I noticed is, the machine was not cutting well enough after mirror allignings and lens changes. As a result of the cutting trials I made at the 4 corners and mid points of the machine bed, I realized that it could works only one corner as well as before. Mirror angles or the angle of the bed also seem a bit distorted.

**Comment by MatthewCroughan** [comment]
@FatihDeveci Is it possible you could upload some of your files for us to play with? Can we also have the source file *before* you convert it to DXF, i.e dwg/svg etc.

6 sounds like fonts being hidden in the file.

5 sounds like it could possibly be genuine noise coming from the USB connection when the file is being uploaded

4 definitely sounds like a mechanical problem, it could easily be that the belts are too tight or too loose as the machine hasn't been maintained in a while.

However I've also seen similar situations when the machine hits a point in a file that has been badly encoded (fonts again, for example) and freaks out, forcing it to overspeed in a given direction, which in turn causes the belt to skip.

3 Sounds like duplicated lines in your file, due to how it has been exported. If you zoom into the problematic line in Laser Cut 5.3, you may see that there is more than one line where you expect there to only be a single line.

If you try to import this DXF file you have into Inkscape, what happens? It should report the error for what it is and give you a preview of what the laser is observing.

**Comment by FatihDeveci** [comment]
@MatthewCroughan I have attached a few files that I tried to cut on Monday. Uploading the files here, I left them as it was on Monday, not even changing their names. I also installed dwg versions. In the past weeks, there have been drawings that I cut without problems and this week I had problems. I've tried a lot of files but have attached a few sample files. Only "05.10.2020.dxf" has dwg version (uploaded). I always worked in dxf format on other files. I also tried to import them to Inskpace today, only "05.10.2020.dxf" gave error.

"a3 witches.dxf": There are 2 groups (top and bottom) of drawings there. I've imported it to the "Laser Cut" program entirely and used it by deleting a group of it. In the top group, the biggest 3 identical shape have shifted. Then I cut the bottom group, there was no problem.

"a3 witches v2.dxf": In this file, I addressed the problematic group in the file I wrote above. I broke up the polylines in 3 large segments. I rotated the file 90 degrees. This way I was able to cut the left group. However, the problem that I described in item number 5 before occurred. Focus of the laser has shifted with big noise just before the finishing beep.

"05.10.2020.dxf": This file is @MatthewCroughan tried on Monday. I had the most difficulty with this file. I have tried this file at least 10 times with different ways. However, the layers shifted every times. I could not complete the cutting. The texts in dwg version of this file were converted to line before converting to dxf. In other words, the text consist of lines that are completely placed side-by-side. Technically, there is no text in my dxf file.

[samples.zip](https://github.com/DoESLiverpool/somebody-should/files/5350704/samples.zip)

**Comment by MatthewCroughan** [comment]
Sean hasn't done anything with the software, but has leveled the jacks, re-tensioned the bed, tightened up the belts on all axis and made a small adjustment to the beam alignment. Seems to be cutting well across the whole bed now.

Some of the above issues from @FatihDeveci are likely to be mechanical, but some others are also sure to be issues with the software stack used.

**Comment by Knottyashcreative** [comment]
I tried to use the laser cutter today but it skipped/slipped and continued into another area.

I tried using different settings and a different parts  of the laser bed. 

None of these methods resolved this error. 


Jackie said there is a laser team , I wonder if I could be part of this as it, would be useful.

**Comment by JackiePease** [comment]
Brett didn't have any issues with the laser cutter this afternoon - perhaps this is another issue relating to polylines.  It would be good to try out a selection of files and see what issues we get.
@Sean-anotherone - Terry (@knottyashcreative) would like to help with laser cutters. Might be worth arranging a chat.

**Comment by Sean-anotherone** [comment]
I agree with the poly lines comment- the machine is tight mechanically so the ropey software does look to be the weak link.  I had issues getting a file from inkscape to import correctly too,  I still need to get to the bottom of the issue. 
Hard to comment further without seeing the problem 1st hand.

Sent from Yahoo Mail on Android 
 
  On Tue, 13 Oct 2020 at 18:06, Jackie Pease<notifications@github.com> wrote:   


Brett didn't have any issues with the laser cutter this afternoon - perhaps this is another issue relating to polylines. It would be good to try out a selection of files and see what issues we get.
@Sean-anotherone - Terry (@Knottyashcreative) would like to help with laser cutters. Might be worth arranging a chat.

**Comment by JackiePease** [comment]
Vicki mentioned that she had had an issue with jumping last night (using previously working files created with Inkscape) and sent these photos:
![IMG-20201017-WA0011](https://user-images.githubusercontent.com/12802256/96365141-65472680-1136-11eb-9a1b-1a64b4801ef0.jpg)
![IMG-20201017-WA0012](https://user-images.githubusercontent.com/12802256/96365143-65dfbd00-1136-11eb-9371-9ff9a1238121.jpg)

**Comment by Sean-anotherone** [comment]
The dodgy bearings and shaft have now been replaced.  X axis now smooth.
Air assist junction repaired too.

**Comment by arthurrowland** [comment]
Has anyone noticed behaviour like this since Nov 2020? Moving to close this issue following replacement of bearings and shaft.

**Comment by Sean-anotherone** [comment]
Still need to do the shaft again as the first one was made from a bit of rebar, should be a ground bearing fit.
Sean
    On Wednesday, 26 May 2021, 15:49:58 BST, arthurrowland ***@***.***> wrote:  
 
 


Has anyone noticed behaviour like this since Nov 2020? Moving to close this issue following replacement of bearings and shaft.

**Comment by FatihDeveci** [comment]
I experienced these problems again, months later, yesterday. Unlike November issues, laser head was stuck and shifted with big noise in simpler cutting moments. Generally, I have seen this problem when cutting only very simple lines of the files that are slightly loaded (takes a little longer to download). As an example, I imported a very detailed drawing. However, laser head was stuck and shifted when it cutting only simple straight lines (in y-axis) in the first cutting layer. I was able to cut the drawings in the file by splitting them into small pieces.

In the last 5 months, I have been able to cut many drawings of the same or more detail and density without any problems.
